src/main/java/net/smartcosmos/dao/relationships/impl/RelationshipPersistenceService.java

```diff
@@ -46,65 +45,106 @@ public RelationshipPersistenceService(
+     *
+     * @param tenantUrn the tenant URN
+     * @param urn the relationship's system-assigned URN
+     * @return
+     */
     @Override
     public List<RelationshipResponse> delete(String tenantUrn, String urn) {
 
-        UUID tenantId = UuidUtil.getUuidFromUrn(tenantUrn);
-
         try {
+            UUID tenantId = UuidUtil.getUuidFromUrn(tenantUrn);
             UUID uuid = UuidUtil.getUuidFromUrn(urn);
             return convertList(relationshipRepository.deleteByTenantIdAndId(tenantId, uuid));
 
-        } catch (Exception e) {
-            exceptionLogger("RelationshipPersistenceService:delete", e, "tenantUrn: " + tenantUrn, "urn: " + urn);
+        } catch (IllegalArgumentException | ConversionException e) {
+            String msg = String.format("findByUrn failed, tenant: '%s', urn: '%s', cause: %s", tenantUrn, urn, e.toString());
+            log.error(msg);
+            log.debug(msg, e);
+            throw e;
         }
-        return new ArrayList<>();
     }
-    // endregion
 
-    // region Find specific / by URN
+    /**
+     * Find a relationship by its URN.
+     *
+     * @param tenantUrn the tenant URN
+     * @param urn the relationship's system-assigned URN
+     * @return
+     */
     @Override
     public Optional<RelationshipResponse> findByUrn(String tenantUrn, String urn) {
 
-        UUID tenantId = UuidUtil.getUuidFromUrn(tenantUrn);
-        Optional<RelationshipEntity> entity = Optional.empty();
-
         try {
+            UUID tenantId = UuidUtil.getUuidFromUrn(tenantUrn);
             UUID uuid = UuidUtil.getUuidFromUrn(urn);
-            entity = relationshipRepository.findByTenantIdAndId(tenantId, uuid);
-        } catch (IllegalArgumentException e) {
-            exceptionLogger("RelationshipPersistenceService:findByUrn", e, "tenantUrn: " + tenantUrn, "urn: " + urn);
-        }
-        if (entity.isPresent()) {
-            final RelationshipResponse response = conversionService.convert(entity.get(), RelationshipResponse.class);
-            return Optional.ofNullable(response);
+            Optional<RelationshipEntity> entity = relationshipRepository.findByTenantIdAndId(tenantId, uuid);
+            if (entity.isPresent()) {
+                final RelationshipResponse response = conversionService.convert(entity.get(), RelationshipResponse.class);
+                return Optional.ofNullable(response);
+            }
+            return Optional.empty();
+
+        } catch (IllegalArgumentException | ConversionException e) {
+            String msg = String.format("findByUrn failed, tenant: '%s', urn: '%s', cause: %s", tenantUrn, urn, e.toString());
+            log.error(msg);
+            log.debug(msg, e);
+            throw e;
         }
-        return Optional.empty();
     }
 
+    /**
+     * Find a particular relationship based on its source, target, and relationshipType.
+     *
+     * @param tenantUrn the tenant URN
+     * @param sourceType the source entity type
```

Comment:
javadoc does not match code.  
- too many params
- searching for sourceUrn right?